source/constructs/api/search/main.py

```diff
@@ -73,25 +74,20 @@ class Query(BaseModel):
 
 @router.get("/property_values", response_model=BaseResponse)
 @inject_session
-def filter_values(table: str, column: str):
+def filter_values(table: str, column: str, condition: str):
+    cond = json.loads(condition) if condition else None
     for searchable_class in searchable:
         if searchable_class.__tablename__ == table:
-            distinct = crud.get_filter_values(searchable_class, column)
+            distinct = crud.get_filter_values(searchable_class, column, cond)
             break
-    # for table_meta in searchable_table['tables']:
-    #     if table_meta['name'] == table_name:
-    #         model_name = table_meta['model']
-    #         break;
     values = []
-    # distinct = crud.get_filter_values(model_name, property)
     for value in distinct:
-        if value[column] is not None:
-            values.append(value[column])
+        if len(value) > 0 and value[0] is not None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

Reply:
Done.